### PR TITLE
SSM Processing Fix

### PR DESCRIPTION
**Pull request by drewjj**
# PR Details
## Description

This resolves issues with how arrays and certain fields were handled in parsing the ASN1 Codec's XML for SSM messages.

The following fields were addressed:
- payload.data.sequenceNumber
- payload.data.status.signalStatus
- payload.data.status.signalStatus.sigStatus.signalStatusPackage

## Related Issue

No related USDOT issue

## Motivation and Context

SSM messages are not currently being processed correctly and result in malformed SSM JSON being written to the topic.OdeSsmJson Kafka topic. This is a problem for all data analysts that need accurate SSM data for their reports.

## How Has This Been Tested?

This has been tested with real SSM data from the CDOT Region 1 snowplows.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/SignalStatusPackageListBuilder.java [hunk 1]
```diff
@@ -14,22 +14,25 @@ private SignalStatusPackageListBuilder() {
...
+                }
+            } else {
+                signalStatusPackageList.getSigStatus()
+                        .add(SignalStatusPackageBuilder.genericSignalStatusPackage(signalStatusPackage));
+    
+            }
+        }
+        else {
+            return null;
+        }
```
Comment (mcook42) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/SignalStatusPackageListBuilder.java:35]: suggestion(non-blocking): invert the if to pull the main code flow out of the if statement and make it clearer to the reader that we return a null value if there isn't a JsonNode for "SignalStatusPackage".
```suggestion
        if (signalStatusPackage == null) {
            return null;
        }
        
        if (signalStatusPackage.isArray()) {
            Iterator<JsonNode> elements = signalStatusPackage.elements();

            while (elements.hasNext()) {
                signalStatusPackageList.getSigStatus()
                        .add(SignalStatusPackageBuilder.genericSignalStatusPackage(elements.next()));
            }
        } else {
            signalStatusPackageList.getSigStatus()
                    .add(SignalStatusPackageBuilder.genericSignalStatusPackage(signalStatusPackage));

        }
```

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/SignalStatusListBuilder.java [hunk 2]
```diff
@@ -14,21 +14,21 @@ private SignalStatusListBuilder() {
...
-			JsonNode signalStatus = status.get("SignalStatus");
-			if(signalStatus != null)
-			{
-				signalStatusList.getStatus()
-					.add(SignalStatusBuilder.genericSignalStatus(signalStatus));
-			}
-		}
+        JsonNode signalStatus = status.get("SignalStatus");
+        if(signalStatus != null)
+		{
```
Comment (mcook42) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/SignalStatusListBuilder.java:19]: polish(non-blocking): consistent if block style
```suggestion
        if(signalStatus != null) {
```

**Review by mcook42 — APPROVED** [review]

**Review by dmccoystephenson — APPROVED** [review]
Looks good & the unit tests pass!

**Review by Michael7371 — APPROVED** [review]
Looks good!

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/SignalStatusListBuilder.java (hunk 2 above)
Comment (drewjj) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/SignalStatusListBuilder.java]: They should now be consistent now.

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/SignalStatusPackageListBuilder.java (hunk 1 above)
Comment (drewjj) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/SignalStatusPackageListBuilder.java]: I reversed the if statement as you requested and applied the same thing to the SignalStatusListBuilder.

**Pull request merged**